scripts/cr2_to_jpg.sh

```diff
@@ -1,10 +1,37 @@
-#!/usr/bin/env bash
+#!/bin/bash -e
+
+usage() {
+  echo -n "##################################################
+# Make a jpeg from the Canon Raw v2 (.CR2) file.
+# 
+# If exiftool is present this merely extracts the thumbnail from
+# the CR2 file, otherwise use dcraw to create a jpeg.
+#
+# If present the CAPTION is added as a caption to the jpeg.
+##################################################
+ $ $(basename $0) FILENAME [CAPTION]
+ 
+ Options:
+  FILENAME          Name of CR2 file that holds jpeg.
+  CAPTION           Optional caption to be placed on jpeg.
+
+ Example:
+  scripts/cr2_to_jpg.sh /var/panoptes/images/temp.cr2 \"M42 (Orion's Nebula)\"
+"
+}
+
+if [ $# -eq 0 ]; then
+    usage
+    exit 1
+fi
 
 FNAME=$1
-NAME=$2
+CAPTION=$2
```

Comment:
Please consider running shellcheck on this and following much of its advice. Here it would probably recommend quoting $2. I would recommend:  `CAPTION="${2}"`

Reply:
I didn't even know of shellcheck! I don't do much actual bash scripting. Thanks!